distributed/diagnostics/plugin.py

```diff
@@ -345,7 +400,50 @@ async def _set_restarted(self, worker):
         )
 
     def _compose_restarted_key(self, worker):
-        return [self.id, "restarted", worker.nanny]
+        return [self.name, "restarted", worker.nanny]
+
+
+class PipInstall(PackageInstall):
+    """A Worker Plugin to pip install a set of packages
```

Comment:
Should we add a deprecation in the class docstring as well? How do we usually do that?